### PR TITLE
Fix and break apart get_apps_in_domain

### PR DESCRIPTION
**Pull request by dannyroberts**
This is a random thing I found when I was surveying view usages. `get_apps_in_domain(full=False)` still fetched and returned the full app. This PR
- splits `get_apps_in_domain` into two functions for full and brief
- modifies test to catch the performance bug
- makes the brief one correctly implement the optimization it purports to

### CODE REVIEW

File: corehq/apps/app_manager/dbaccessors.py [hunk 1]
```diff
@@ -103,30 +103,29 @@ def get_app(domain, app_id, wrap_cls=None, latest=False, target=None):
...
-        view_name = 'app_manager/applications_brief'
-        startkey = [domain]
-        endkey = [domain, {}]
 
+def get_brief_apps_in_domain(domain, include_remote=True):
     from .models import Application
-    view_results = Application.get_db().view(view_name,
-        startkey=startkey,
-        endkey=endkey,
-        include_docs=True)
```
Comment (dannyroberts) [inline corehq/apps/app_manager/dbaccessors.py]: It's pretty hard to tell just looking at the original code, but here is the root of the performance bug. (It's very minor in the grand scheme of things, but definitely wrong.) It was calling `applications/applications_brief` with `include_docs=true`, which entirely defeats the purpose of calling that view.

**Comment by dannyroberts** [comment]
@nickpell

**Comment by nickpell** [comment]
looks good pending tests

**Comment by dannyroberts** [comment]
Test failures pointed some other problems out to me which I'm fixing.

File: corehq/apps/domain/models.py [hunk 2]
```diff
@@ -681,7 +681,7 @@ def get_latest_app_id(doc_id):
                 if app:
                     return app._id, app.doc_type
 
-            for app in get_apps_in_domain(self.name):
+            for app in get_brief_apps_in_domain(self.name):
```
Comment (dannyroberts) [inline corehq/apps/domain/models.py]: It turns out this is the only place (besides tests) that actually wanted to call the brief version. By default full=False, but all other callers were relying on the behavior that made it equivalent to full=True. (I had to read the context of each call to figure that out.)

**Comment by dannyroberts** [comment]
I think it'd be irresponsible to submit this without tests for dbaccessors, so I'm going to do that and then I think it's g2m

File: corehq/apps/app_manager/tests/test_brief_view.py [hunk 3]
```diff
@@ -1,32 +0,0 @@
...
-            app.delete()
-        cls.project.delete()
-
-    def test_domain(self):
-        apps = self.project.applications()
-        self.assertEqual(len(apps), 2)
-
-    def test_app_base(self):
-        apps = get_apps_in_domain(self.domain)
-        self.assertEqual(len(apps), 2)
```
Comment (dannyroberts) [inline corehq/apps/app_manager/tests/test_brief_view.py]: The new tests make these redundant

File: corehq/apps/app_manager/tests/__init__.py [hunk 4]
```diff
@@ -1,44 +1,44 @@
...
+    from corehq.apps.app_manager.tests.test_bulk_ui_translation import *
     from corehq.apps.app_manager.tests.test_case_list_form import *
-    from corehq.apps.app_manager.tests.test_form_workflow import *
     from corehq.apps.app_manager.tests.test_case_list_lookup import *
+    from corehq.apps.app_manager.tests.test_case_meta import *
     from corehq.apps.app_manager.tests.test_child_module import *
-    from corehq.apps.app_manager.tests.test_report_config import *
+    from corehq.apps.app_manager.tests.test_commcare_settings import *
+    from corehq.apps.app_manager.tests.test_days_ago_migration import *
+    from corehq.apps.app_manager.tests.test_dbaccessors import *
```
Comment (dannyroberts) [inline corehq/apps/app_manager/tests/__init__.py]: Sorry, I got OCD and sorted these. I just added this import and removed test_brief_view.

**Comment by dannyroberts** [comment]
:white_check_mark:

File: corehq/apps/app_manager/tests/test_dbaccessors.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+            Application(domain='decoy-domain')
+        ]
+        for app in cls.decoy_apps:
+            app.save()
+
+    @classmethod
+    def tearDownClass(cls):
+        for app in cls.apps + cls.decoy_apps:
+            app.delete()
+        # to circumvent domain.delete()'s recursive deletion that this test doesn't need
```
Comment (czue) [inline corehq/apps/app_manager/tests/test_dbaccessors.py]: this is smart

File: corehq/apps/prelogin [hunk 6]
```diff
@@ -1 +1 @@
-Subproject commit 962b5dad339911c9f5d93f2d87d03a22d3ebe9a0
+Subproject commit fe7abbeb0b1cd3caba9053df8d7d99e5775b699a
```
Comment (czue) [inline corehq/apps/prelogin]: is this supposed to be here?

File: corehq/apps/app_manager/tests/test_dbaccessors.py [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+        apps = get_apps_in_domain(self.domain)
+        self.assertEqual(len(apps), 2)
+        normal_app, remote_app = sorted(apps, key=lambda app: app.is_remote_app())
+        expected_normal_app, expected_remote_app = sorted(self.apps, key=lambda app: app.is_remote_app())
+        self.assert_docs_equal(remote_app, expected_remote_app)
+        self.assert_docs_equal(normal_app, expected_normal_app)
+
+    def test_domain_has_apps(self):
+        self.assertEqual(domain_has_apps(self.domain), True)
+        self.assertEqual(domain_has_apps('somecrazydomainthathasnoapps'), False)
```
Comment (czue) [inline corehq/apps/app_manager/tests/test_dbaccessors.py]: thoughts on adding a test to exercise `include_remote=False`? seems like it'd be super easy and possibly valuable

**Comment by czue** [comment]
not merging in case prelogin change is accidental but g2g once that's confirmed

File: corehq/apps/prelogin (hunk 6 above)
Comment (dannyroberts) [inline corehq/apps/prelogin]: oh, no it's not. Let me fix that

**Comment by dannyroberts** [comment]
OK added tests (and otherwise made no changes). The tests pass locally, so I think no need to wait for tests but also no particular rush

**Pull request merged**